### PR TITLE
US132833 Improvements to release date component

### PR DESCRIPTION
**Pull request by dustxd**
### Summary of Changes
* Updated lang term from `Release Date` to `Display Date` as per design
* Disabled the Release Date calendar input when permissions are missing
* Stopped sending PATCH requests when calendar input value hasn't changed
  * Context: to make a change in the calendar input, the `@change` event is triggered twice
    * Once on initial click
    * Once on the actual value change
 * Updated the `updateProperty()` function in quiz submission view mobx state to coordinate updates from the API and the optimistic UI changes - this fixes the flickering issues for the checkboxes in the dialog as well
   * Context: The flickering happens when the first PATCH request (for working copy) isn't done yet and the user quickly clicks again. The component will update to the newest value but flicker to the less new value as the first PATCH request returns and flicker back to the newest value once the second PATCH request returns => `this._entity = entity` instead of `this.load(entity)` fixes this issue.

### Rally Story
https://rally1.rallydev.com/#/?detail=/userstory/611715032115&fdp=true

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js [hunk 1]
```diff
@@ -88,17 +88,20 @@ class ActivityQuizSubmissionViewsAccessRestrictionsEditor
 			canUpdateTimeLimitNumber,
 			timeLimitNumber,
 			gradeRestrictions,
-			releaseDate
+			releaseDate,
+			canUpdateReleaseDate
```
Comment (mostlyfabulous) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js:92]: I think we need to check `entity` is not undefined, the usual boilerplate checks we have 😄 in our code.

Comment (dustxd) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js:92]: Good point

**Review by mostlyfabulous — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-quiz-editor/state/quiz-submission-view.js [hunk 2]
```diff
@@ -213,7 +213,7 @@ export class QuizSubmissionView {
 
 	async updateProperty(updateFunc) {
 		this.saving = updateFunc();
```
Comment (stevenao) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/state/quiz-submission-view.js:215]: we usually have the `await` at the function call level. `this.saving = await updateFunc();`

Comment (dustxd) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/state/quiz-submission-view.js:215]: We're actually doing that on the next line 😄 `await this.saving;` is equivalent to `await updateFunc();`

Comment (stevenao) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/state/quiz-submission-view.js:215]: Ok. Just a nit. The saving var is holding a promise. I thought it was a value variable.

**Pull request merged**